Infer source group(s) from user/token group membership

acrellin opened the pull request:
Infer source group(s) from user/token group membership if IDs not explicitly specified in POST handler

stefanv commented on skyportal/handlers/source.py:
```diff
@@ -114,7 +114,12 @@ def post(self):
         try:
             group_ids = data.pop('group_ids')
```
Is there any verification done to ensure that a user doesn't add sources to a group they don't belong to?

stefanv commented on skyportal/handlers/source.py:
```diff
@@ -114,7 +114,12 @@ def post(self):
+            if self.current_user.groups:
+                group_ids = [g.id for g in self.current_user.groups]
+            else:
+                return self.error("group_ids is a required field when user/token "
```
I can't quite parse this error message

acrellin commented:
@stefanv thanks for the feedback. I've added a check to ensure a user cannot add sources to groups to which they don't belong, and have also changed the error message.

stefanv commented on skyportal/handlers/source.py:
```diff
@@ -111,10 +111,14 @@ def post(self):
+                              "you can add sources.")
         try:
-            group_ids = data.pop('group_ids')
+            group_ids = [id for id in data.pop('group_ids') if id in user_group_ids]
```
What happens when `group_ids` is empty?

If the group is added to all user groups by default, we should document that.

acrellin replied:
Okay -- document that where?  In the docstring description?

acrellin replied:
Do you mean what happens if a user specifies `group_ids = []` in the request data?  In that case, it returns with an error message.

stefanv replied:
No, more the situation where the user specifies group IDs that either don't exist or where the user doesn't have permission to post to, so that we end up with no valid IDs.

Documentation should go with the endpoint docs, so it appears in our API docs.

acrellin replied:
In that case, we return with the error message "Invalid group_ids field. Please specify at least one valid group ID that you belong to."

acrellin commented:
@stefanv good to go here?
